src/pal/inc/pal_assert.h

```diff
@@ -33,6 +33,21 @@ extern "C" {
 #endif // __cplusplus
 
 #if defined(_DEBUG)
+
+#ifdef __GNUC__
+
+#define _ASSERTE(e) do {                                        \
```

Comment:
I don't think we should fix the original issue this way. Now the assert just prints the error and the code continues running. This macro is used in the whole coreclr runtime.
I can see two possible solutions:
1. Let the profiler developers define these two functions. They can be just dummy since profiler developers care only for the data structures.
2. Define the _ASSERTE here as empty (like for release build) when the pal.h header is included from a profiler code. We would need to come up with some #define that would control that (unless there is already one).

Reply:
@janvorli Hi! In the first case, we should define these functions globally, and therefore someone may accidentally start using it in our code base on Linux/MacOs, and it won't be a compile error. I dislike it for that reason. The second option looks better from this point of view.

> Now the assert just prints the error and the code continues running.

Have you thought about smth like `__builtin_debugtrap()` after `fprintf()`? Why it doesn't satisfy you?

Reply:
The `__builtin_debugtrap` is defined by clang only, so it would not work for GCC. There was a recent effort to make coreclr runtime buildable by GCC too.

Reply:
While it seems we could possibly use what GCToOSInterface::DebugBreak() in gcenv.unix.cpp does to implement DebugBreak, but the problem here is that we cannot include any platform headers from code that is part of runtime (the gcenv.unix.cpp is not). So we cannot include headers containing the `raise` or `SIGTRAP`.

So my recommendation would be to do it the way @jkotas suggested.

Reply:
You can use `__asm__ __volatile__ ("int3")` on x86 and x86_64 to implement a debug trap without requiring any system headers.  There might be similar things in other arches, too.